src/utils.js

```diff
@@ -362,6 +400,26 @@ export const changePin = async (wallet, oldPin, newPin) => {
   return true;
 };
 
+/**
+ * @param {string} oldPin
+ * @param {string} newPin
+ *
+ * @return {Promise<boolean>} Wether the change password was successful
+ */
+export async function changePinOnAccessData(oldPin, newPin) {
```

Comment:
Why do you need this new method? Can't you use the method `changePin` above? The code seems the same, you just need the wallet object.

Reply:
Yes, but to create the wallet object we need to have a started wallet and this is not the case when the user unlocks the wallet and we need to migrate the access data before starting the wallet.

Also, migration after the wallet is started can be done but may be bad to the UX because of the order of operations:
- Input pin to unlock wallet
- Load history and show list of transactions
- Migrate access data from the started wallet
  - CPU intensive with no UI indicator.
  - Wallet may be unresponsive.

Reply:
Why don't you change the `changePin` method to use this one? You can get the `storage` from the `wallet` object and call this new method, which would receive the `storage` object.

Does it make sense? I'm trying to reduce the duplicated code here.

Reply:
I didn't because of the `checkPinAndPassword` method of the wallet that is called.
But I can use the storage methods for this

Reply:
Done